pkg/controller/apicuriodeployment/apicuriodeployment_controller.go

```diff
@@ -0,0 +1,257 @@
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileApiCurioDeployment{
+		client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+		config: mgr.GetConfig(),
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	opts := controller.Options{Reconciler: r}
+	c, err := controller.New("apicuriodeployment-controller", mgr, opts)
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource ApicurioDeployment
+	err = c.Watch(&source.Kind{Type: &integreatlyv1alpha1.ApicurioDeployment{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner ApicurioDeployment
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &integreatlyv1alpha1.ApicurioDeployment{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileApiCurioDeployment{}
+
+
+
+func (r *ReconcileApiCurioDeployment) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling ApicurioDeployment %s/%s\n", request.Namespace, request.Name)
+
+	// Fetch the ApicurioDeployment instance
+	instance := &integreatlyv1alpha1.ApicurioDeployment{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+
+		return reconcile.Result{}, err
+	}
+
+
+	err = r.bootstrap(request)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	exts, err := r.processTemplate(instance)
+	if err != nil {
+		logrus.Errorf("Error while processing template: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	objs, err := r.getRuntimeObjects(exts)
+	if err != nil {
+		logrus.Errorf("Error while retrieving runtime objects: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	err = r.createObjects(objs, request.Namespace, instance)
+	if err != nil {
+		logrus.Errorf("Error creating runtime objects: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileApiCurioDeployment) bootstrap(request reconcile.Request) error {
+	var err error
+
+	if r.tmpl == nil {
+		restCfg := r.config
+		namespace := request.Namespace
+		r.tmpl, err = openshift.NewTemplate(namespace, restCfg, openshift.TemplateDefaultOpts)
+
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileApiCurioDeployment) processTemplate(cr *integreatlyv1alpha1.ApicurioDeployment) ([]runtime.RawExtension, error) {
+	tmplPath := cr.Spec.Template
+	if tmplPath == "" {
+		return nil, fmt.Errorf("Spec.Template.Path property is not defined")
+	}
+
+
+	res, err := openshift.LoadKubernetesResourceFromFile(BaseTmplPath + "/" + tmplPath)
```

Comment:
Any thoughts on using packr https://github.com/gobuffalo/packr to embed the template in the binary rather than going down the route of modifying the Dockerfile?

Is there any advantage to being able to override the template in the spec? Is it even possible to load a template for a different template that hasn't been built into the image?

Reply:
I don't see any advantage in adding the templates to the docker file other than being able to change them in the running container (which you probably shouldn't anyway). So i'd be in favor of using packr or a similar way of adding them to the binary.

Reply:
This seems like a reasonable approach and saves have to read files of disk etc